### PR TITLE
Added UrlEmbedService so that modules do not have to call the library directly and shared config can be used site-wide.

### PR DESCRIPTION
**Pull request by davereid**

### CODE REVIEW

**Comment by davereid** [comment]
This will allow us to start adding caching to embeds and make other things possible.

File: src/UrlEmbedHelperTrait.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ * @file
+ * Contains Drupal\url_embed\UrlEmbedHelperTrait.
+ */
+
+namespace Drupal\url_embed;
+
+use Drupal\Core\Extension\ModuleHandlerInterface;
+
+/**
+ * Wrapper methods for URL embedding..
```
Comment (cs-shadow) [inline src/UrlEmbedHelperTrait.php]: nit: multiple periods at the end

File: src/UrlEmbedService.php [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    return $this->config;
+  }
+
+  public function setConfig(array $config) {
+    $this->config = $config;
+  }
+
+  /**
+   * @param string|\Embed\Request $request The url or a request with the url
+   * @param array          $config  Options passed to the adapter
```
Comment (cs-shadow) [inline src/UrlEmbedService.php]: nit: extra spacing after `@param`

**Comment by cs-shadow** [comment]
Couple of nitpicks. Looks good otherwise.

**Pull request merged**